### ISSUE TITLE
Using @ts-expect-error on class method signature would disable checking all other methods in interface

### ISSUE THREAD

**Issue by fenying** [microsoft/TypeScript#59266]
### 🔎 Search Terms

ts-expect-error missing method

### 🕗 Version & Regression Information

- This is the behavior in every version I tried between v4.0 and v5.5, didn't try older versions

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.5.3#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgSQBVgBneAbwChK5a5MAKASgC44A3CBAEwG5q6cAEbM2nHvwC+1UJFhxsAG0wkScYmUQBbMIuBbgKNUVIUBdAPQW4AARgkAtLODYYTqFGg06YAK5DFBGx6BgBrNjIoZABzVg4ubjhyaWkgA

### 💻 Code

```ts
export interface ITest {

    a(): void;

    b(): void;
}

export class Test implements ITest {

    // @ts-expect-error
    public a(k: string): void {}

    // method b is missing, but no errors reported
}
```

### 🙁 Actual behavior

No error reported even missing a method in the class

### 🙂 Expected behavior

Should report errors if any method is not implemented in class

**Comment by MartinJohns** [comment]
`ts-expect-error` is not the culprit here, you also don't get the error you expect when you remove it. The compiler doesn't check if your class implements the interface when there's already an error inside your class. Using `ts-expect-error` does not remove the error and let the compiler happily continue its job, it just *hides* the error.

Related comment: https://github.com/microsoft/TypeScript/issues/47551#issuecomment-1018994226

>  once your program has a type error in it, it's up to our heuristics to decide exactly what the most useful reporting of that type error is.

**Comment by fenying** [comment]
OH. I see, the compiler stops checking next error if it meet an error, until this one got resolved :(
